File: assignments/cash_counting/javascript/cash_counting_test_runner.js
const assert = require('assert');
const { CashCounting } = require('./cash_counting.js'); // Assuming student file is Cash.js

function runTests() {
    const solver = new CashCounting();
    // Corrected Test Cases: { amount: number, expected: number }
    const testCases = [
        { amount: 70, expected: 3 }, { amount: 63, expected: 5 }, { amount: 47, expected: 5 },
        { amount: 268, expected: 6 }, { amount: 10000, expected: 1 }, { amount: 15, expected: 1 },
        { amount: 14, expected: 5 }, { amount: 29, expected: 6 }, { amount: 0, expected: 0 },
        { amount: 1, expected: 1 }, { amount: 4, expected: 4 }, { amount: 5, expected: 1 },
        { amount: 9, expected: 5 }, { amount: 10, expected: 1 }, { amount: 12345, expected: 8 }
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for Cash Change (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual = -1;
        const caseIdentifier = `Test Case #${i + 1}: Input amount=${tc.amount}`;
        try {
            actual = solver.calculateNotes(tc.amount);
            assert.strictEqual(actual, tc.expected);
            report.push(`${caseIdentifier} -> ✅ PASS`);
            passedCount++;
        } catch (error) {
             report.push(`${caseIdentifier} -> ❌ FAIL/ERROR: Expected ${tc.expected}, but got ${actual}`);
        }
    });

    console.log(report.join('\n\n'));
    console.log("\n===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();

File: assignments/cash_counting/python/cash_counting_test_runner.py
import sys
from cash_counting import CashCounting # Import the student's class

def run_tests():
    solver = CashCounting()
    # Corrected Test Cases: (input_amount, expected_note_count)
    test_cases = [
        (70, 3), (63, 5), (47, 5), (268, 6), (10000, 1),
        (15, 1), (14, 5), (29, 6), (0, 0), (1, 1),
        (4, 4), (5, 1), (9, 5), (10, 1), (12345, 8)
    ]
    passed_count = 0
    report = []
    print("===== Starting Automated Tests for Cash Change (Python) =====")

    for i, (input_amount, expected) in enumerate(test_cases):
        actual = -1
        case_identifier = f"Test Case #{i + 1}: Input amount={input_amount}"
        try:
            actual = solver.calculate_notes(input_amount)
            report.append(case_identifier)
            report.append(f" -> Expected: {expected}, Got: {actual}")

            if actual == expected:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(case_identifier)
            report.append(f" -> ❌ ERROR: An exception occurred - {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary ====="); print(f"Total Tests: {len(test_cases)}"); print(f"Passed: {passed_count}"); print(f"Failed: {len(test_cases) - passed_count}"); print("========================")
    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()